### PR TITLE
Make k8s.io/docs/home to support i18n

### LINKED ISSUE #12529: k8s.io/docs/home does not support i18n
**This is a...** 

- [ ] Feature Request
- [ ] Bug Report

**Problem:**

Currently [k8s.io/docs/home](https://kubernetes.io/docs/home/) page provides a good [RoadMap](http://wiki.c2.com/?RoadMap) for navigating the K8s docs. But it is hard-coded to support `en` contents only. 

https://github.com/kubernetes/website/blob/6224377d8796f50f4d7791112ccda89e67bf7a4b/layouts/docs/docsportal_home.html#L5

For comparison:
* en: https://kubernetes.io/docs/home/
* zh: https://kubernetes.io/zh/docs/home/
* ko: https://kubernetes.io/ko/docs/home/

**Proposed Solution:**

I think there're two possible solutions for this issue. 

1. Convert the links and titles automatically by [using multilingual feature of Hugo](https://gohugo.io/content-management/multilingual/#reference-the-translated-content). But AFAIK, this can not cover the anchored links (ex. "/docs/setup/pick-right-solution/#hosted-solutions")
2. Convert the links and titles manually by appending `if` branch per `.Lang` to the `/layouts/docs/docsportal_home.html`

**Page to Update:**
* zh: https://kubernetes.io/zh/docs/home/
* ko: https://kubernetes.io/ko/docs/home/

### PR DESCRIPTION
**Pull request by gochist**
fixes #12529

/cc @chenopis @lucperkins @zacharysarah

### CODE REVIEW

**Comment by zacharysarah** [comment]
@gochist 👋 

I see how this fixes #12529 by moving content out of the hard-coded layout and into `content/en/docs/home/_index.md`. 👍 

The `{{ template "docs-portal-card" . }}` looks like a really elegant improvement to the existing layout. ✨ 

Do I correctly understand that this PR will allow follow-up PRs for the front matter of content in other localizations? For example, will localized files be able to add the following attributes to their front matter?
```
card: 
   name:
   weight:
   title:
```

The [preview for English](https://deploy-preview-12570--kubernetes-io-master-staging.netlify.com/docs/home/) looks good. 

/lgtm

Changes are large enough that I'd like to get a second reviewer to approve. @Bradamant3, @lucperkins, @kbarnard10, your input would be appreciated. 🙇

**Comment by zacharysarah** [comment]
@kubernetes/sig-docs-ja-owners 
@kubernetes/sig-docs-zh-owners 

@sieben @lledru This will also affect French content.

**Comment by gochist** [comment]
@zacharysarah Thank you for reviewing! Yes, you understand this PR correctly.

**Comment by zparnold** [comment]
/lgtm
@zacharysarah I'll wait on any communication you want to do to our other languages for approval 😄

File: layouts/docs/docsportal_home.html [hunk 1]
```diff
@@ -2,144 +2,48 @@
...
 	 <h1>{{ .Title }}</h1>
     {{ end }}
-    {{ if eq .Lang "en" }}
     {{ template "docs-portal-cards" . }}
-    {{ end }}
 {{ end }}
 
 {{ define "content-id" }}content{{ end }}
 
 {{ define "docs-portal-cards" }}
```
Comment (kbhawkey) [inline layouts/docs/docsportal_home.html]: @gochist, Looks good.
Do you know if the `content-id` define is required? 
Suggestion: I initially was confused by the similar template names, `docs-portal-cards` and `docs-portal-card`.
Could rename `docs-portal-cards` to something like, `docs-portal-content` and rename the template that renders all the cards to `docs-portal-cards`.

Comment (gochist) [reply layouts/docs/docsportal_home.html]: @kbhawkey Thank you! I addressed your review. I'm not sure what the `content-id` is for. But when I delete it, the page layout is broken.

Comment (zacharysarah) [reply layouts/docs/docsportal_home.html]: @kbhawkey Great feedback--I had the same confusion. 👍

**Comment by zacharysarah** [comment]
Heads up, folks: this PR changes how we handle content in cards on the docs landing page, and how we handle content to which those card link. (See #12696 for required actionable followup)

This is an upstream change that affects all localizations for 1.13 and following.

@kubernetes/sig-docs-en-owners 
@kubernetes/sig-docs-fr-owners 
@kubernetes/sig-docs-ja-owners 
@kubernetes/sig-docs-zh-owners 
@micheleberardi @rlenferink @Bradamant3 @jaredbhatti 

/approve

**Comment by zacharysarah** [comment]
@jimangel 👋 I recommend merging this into `dev-1.14` and `release-1.13` sooner rather than later, especially if there are any future changes to card content on the docs landing page. Thanks in advance!

**Comment by Rajakavitha1** [comment]
@zacharysarah looks like you may also need to add lgtm to this PR.

**Comment by Rajakavitha1** [comment]
@zacharysarah could you also please state your approval on https://github.com/kubernetes/website/pull/12752#pullrequestreview-206243471 that addresses https://github.com/kubernetes/website/issues/12696

**Pull request merged**